pallets/settlement/src/benchmarking.rs

```diff
@@ -823,10 +823,10 @@ benchmarks! {
 
     affirm_instruction {
 
-        let l in 2 .. T::MaxLegsInInstruction::get() as u32; // At least 2 legs needed to achieve worst case.
+        let l in 0 .. T::MaxLegsInInstruction::get() as u32;
         let (portfolios_to, _, to, _, _, _) = setup_affirm_instruction::<T>(l);
         let instruction_id = 1; // It will always be `1` as we know there is no other instruction in the storage yet.
-    }: _(RawOrigin::Signed(to.account), instruction_id, portfolios_to.clone())
+    }: _(RawOrigin::Signed(to.account), instruction_id, portfolios_to.clone(), (l / 2).into())
```

Comment:
Might move division and type conversion out of the metered block into the setup phase.

Reply:
Oh, and cloning too.

Reply:
My assumption is that it will only start calculating the time difference once it computes all of its arguments.

Reply:
The assumption is not correct. Anything after `:` and until `verify` is first macro-expanded into a `block` AST fragment, as if you had written `: { ... }` and then that `{ ... }` becomes the benchmark to execute. Anything that happens within the `{ ... }` thus contributes to the weight.

Reply:
can you refer me to the documentation @Centril?

Reply:
Here: https://substrate.dev/rustdocs/v2.0.0/frame_benchmarking/macro.benchmarks.html
Its not documented though; you need to follow the macro implementation to realize this.